vent/common/fashion.py

```diff
@@ -0,0 +1,54 @@
+""" Decorators for dangerous functions
+
+"""
+from vent.common.values import MAX_STACK_DEPTH
+from vent.common.logging import log_exception
+import traceback
+import functools
+
+
+def locked(func):
+    """
+    Wrapper to use as decorator, handle lock logic for a
+    @property
+
+    Arguments:
+        func (callable): function to wrap
+    """
+
+    # define a function to return that wraps the inner
+    # function around some standardized error handling
+    # logic
+
+    # use .wraps to preserve inner method identity
+    @functools.wraps(func)
+    def lock_wrapper(self, *args, **kwargs):
+        ret = None
+
+        try:
+            self.lock.acquire()
+            ret = func(self, *args, **kwargs)
+        except Exception as e:
+            log_exception(
```

Comment:
What upper layer exception handling mechanism do you expect? E.g., restart the program, or just retry the reading?

Reply:
i think this is something that the coordinator would handle -- exceptions would trigger different actions depending on the state of the system. eg. if we have some max duration to wait for a new sensor reading, missing a reading below that duration shouldn't trigger anything but after that duration should throw an alarm.

Reply:
Yeah, this is a good example~

Reply:
We definitely have to think through the different exceptions and what actions can/should be taken for each. Another one that comes to mind is when the Pigpio daemon can't be reached, the coordinator should kill and restart it.

Reply:
idk why we should overcomplicate it more than defining a few exception types and mapping them to a few predefined actions -- like make a `PigpioConnectionException` and throw it up to the coordinator who matches exception classes to actions.